Clear key state on disable to handle Shift-up correctly.

diff -urN ../mozc.orig/unix/ibus/mozc_engine.cc ./unix/ibus/mozc_engine.cc
--- ../mozc.orig/unix/ibus/mozc_engine.cc	2012-08-31 14:36:43.000000000 +0900
+++ ./unix/ibus/mozc_engine.cc	2012-11-29 16:56:09.216754385 +0900
@@ -285,6 +285,7 @@
 void MozcEngine::Disable(IBusEngine *engine) {
   RevertSession(engine);
   GetCandidateWindowHandler(engine)->Hide(engine);
+  key_event_handler_->Clear();
 }
 
 void MozcEngine::Enable(IBusEngine *engine) {
